submissions/YuraZagor/Frogger/app.js

```diff
@@ -0,0 +1,108 @@
+const allEnemies = [];
+const speedLow = 200;
+const speedMedium = 275;
+const speedHigh = 350;
+const allSpeeds = [speedLow, speedMedium, speedHigh];
+const enemyRestartX = -55;
+const collisionOffset = 40;
+const cellX = 101;
+const cellY = 83;
+const spriteRepositioningY = -10;
+const playerStartX = 2*cellX;
+const playerStartY = 5*cellY + spriteRepositioningY ;
+const fieldWidth = 5*cellX;
+const fieldHight = 5*cellY;
+const rangesY = [1, 2, 3].map(item => item*cellY+spriteRepositioningY);
+
+function Character (x, y, sprite) {
+    this.x = x;
+    this.y = y;
+    this.sprite=sprite;
+};
+
+Character.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y) 
+};
+
+function Enemy(x, y, speedX, player) {
+    Character.call(this, x, y, 'images/enemy-bug.png');
+    this.speedX = speedX;
+    this.player = player;
+};
+Enemy.prototype = Object.create(Character.prototype);
+
+Enemy.prototype.update = function(dt){
+    this.x += this.speedX * dt;
+    if (this.x > fieldWidth){
+        this.x = enemyRestartX;
+        this.speedX = allSpeeds[Math.floor(Math.random()*3)];
+    };
+    this.checkColision()
+};
+Enemy.prototype.checkColision = function(){ 
+    if (this.y === this.player.y) {
+
+        if ((this.x - this.player.x < collisionOffset ) && (this.player.x - this.x < collisionOffset )) {
+            this.player.x = playerStartX;
+            this.player.y = playerStartY;
+        };    
+    }; 
+};
+
+function addBug() {
+    rangesY.forEach ( (levelMark)=>  allEnemies
+    .push( new Enemy(0, levelMark, allSpeeds[Math.floor(Math.random()*3)], player)) 
+    )
+};
```

Comment:
This wasn't a part of the requirements. But try to scale that narrow requirement onto a bigger scale.
A well designed function relies only on parameters it receives and doesn't refer to any global variables.
This way it encapsulates business logic it designated to implement.
In big codebases following this principle will pay.